### PR TITLE
Support yVault assets for Price Oracle

### PR DESCRIPTION
**Pull request by jeremyHD**

### CODE REVIEW

File: contracts/PriceOracleProxy.sol [hunk 1]
```diff
@@ -33,34 +41,39 @@ contract PriceOracleProxy is PriceOracle, Exponential {
...
 
     /**
      * @param admin_ The address of admin to set aggregators
      * @param v1PriceOracle_ The address of the v1 price oracle, which will continue to operate and hold prices for collateral assets
      * @param cEthAddress_ The address of cETH, which will return a constant 1e18, since all prices relative to ether
      * @param cUsdtAddress_ The address of cUSDC
      * @param cUsdtAddress_ The address of cUSDT
+     * @param cYcrvAddress_ The address of cYcrv
+     * @param cYYcrvAddress_ The address of cYYcrv
+     * @param cYEthAddress_ The address of cYYcrv
```
Comment (StanleyDing) [inline contracts/PriceOracleProxy.sol:59]: cYeth

File: contracts/PriceOracleProxy.sol [hunk 2]
```diff
@@ -75,31 +88,77 @@ contract PriceOracleProxy is PriceOracle, Exponential {
...
+            return yVaultPrice;
+        }
+        
+        if (cTokenAddress == cYcrvAddress || cTokenAddress == cYYcrvAddress) {
+            MathError mathErr;
+            Exp memory price;
+            Exp memory usdEthPrice;
+            Exp memory yCrvPrice;
+
+            (mathErr, price) = getPriceFromChainlink(AggregatorInterface(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419));
```
Comment (StanleyDing) [inline contracts/PriceOracleProxy.sol:106]: // ethUsd price

File: contracts/PriceOracleProxy.sol [hunk 3]
```diff
@@ -75,31 +88,77 @@ contract PriceOracleProxy is PriceOracle, Exponential {
...
+                // Fallback to v1 PriceOracle
+                return getPriceFromV1(cTokenAddress);
+            }
+            
+            // ETH/USD price decimal is 8
+            (mathErr, usdEthPrice) = getExp(1e8, price.mantissa);
+            uint virtualPrice = YSwapInterface(0x45F783CCE6B7FF23B2ab2D70e416cdb7D6055f51).get_virtual_price();
+            
+            (mathErr, yCrvPrice) = mulExp(usdEthPrice, Exp({mantissa: virtualPrice}));
+            if (mathErr != MathError.NO_ERROR || isZeroExp(yCrvPrice)) {
```
Comment (bun919tw) [inline contracts/PriceOracleProxy.sol:117]: isZeroExp might be useless

File: contracts/PriceOracleProxy.sol [hunk 4]
```diff
@@ -75,31 +88,77 @@ contract PriceOracleProxy is PriceOracle, Exponential {
...
+                // Fallback to v1 PriceOracle
+                return getPriceFromV1(cTokenAddress);
+            }
+            
+            if (cTokenAddress == cYYcrvAddress) {
+                uint yVaultPrice = YVaultInterface(0x5dbcF33D8c2E976c6b560249878e6F1491Bca25c).getPricePerFullShare();
+                Exp memory yyCrvPrice;
+                (mathErr, yyCrvPrice) = mulExp(yCrvPrice, Exp({mantissa: yVaultPrice}));
+                if (mathErr != MathError.NO_ERROR || isZeroExp(yyCrvPrice)) {
```
Comment (bun919tw) [inline contracts/PriceOracleProxy.sol:126]: ditto

**Pull request merged**